src/init.cu

```diff
@@ -604,6 +610,7 @@ static ncclResult_t initTransportsRank(struct ncclComm* comm, ncclUniqueId* comm
     return ncclInternalError;
```

Comment:
We should add `free(rankInfos);` here as well (before return ncclInternalError).

Reply:
Done, thanks for spotting that!